pkg/config/config.go

```diff
@@ -118,6 +118,56 @@ func NewMicroshiftConfig() *MicroshiftConfig {
 	}
 }
 
+// Determine if the config file specified a NodeName (by default it's assigned the hostname)
+func (c *MicroshiftConfig) IsDefaultNodeName() bool {
+	hostname, err := os.Hostname()
+	if err != nil {
+		klog.Fatalf("Failed to get hostname %v", err)
+	}
+	return c.NodeName == hostname
+}
+
+// Read or set the NodeName that will be used for this MicroShift instance
+func (c *MicroshiftConfig) establishNodeName() (string, error) {
+	filePath := filepath.Join(GetDataDir(), "nodename")
+	contents, err := os.ReadFile(filePath)
+	if os.IsNotExist(err) {
+		if err := os.WriteFile(filePath, []byte(c.NodeName), 0444); err != nil {
+			return "", fmt.Errorf("failed to write nodename file %q: %v", filePath, err)
+		}
+		return c.NodeName, nil
+	} else if err != nil {
+		return "", err
+	}
+	return string(contents), nil
+}
+
+const (
+	IS_DEFAULT_NODENAME     = true
+	IS_NOT_DEFAULT_NODENAME = false
+)
+
+// Validate the NodeName to be used for this MicroShift instances
+func (c *MicroshiftConfig) ValidateNodeName(isDefaultNodeName bool) error {
```

Comment:
No need to have this method exported if only called from config package.
Unless we have a reason ever to call this method without `IsDefaultNodeName()`, I'd simply initialise that var from within the method.

Reply:
no, that is for testability of the function, but I will make it private.

Reply:
python monkey patching for testing made things easier, but here we cannot do that.